includes/libraries/parsecsv.lib.php

```diff
@@ -0,0 +1,1163 @@
+
+        return $this->_wfile($file, $this->unparse($data, $fields, $append, $is_php), $mode);
+    }
+
+    /**
+     * Output
+     * Generate a CSV based string for output.
+     *
+     * @access public
+     * @param  [string] $filename  If specified, headers and data will be output directly to browser as a downloable file
+     * @param  [array]  $data      2D array with data
+     * @param  [array]  $fields    Field names
+     * @param  [type]   $delimiter delimiter used to separate data
+     *
+     * @return [string]
+     */
+    public function output($filename = null, $data = array(), $fields = array(), $delimiter = null) {
+        if (empty($filename)) {
+            $filename = $this->output_filename;
+        }
+
+        if ($delimiter === null) {
+            $delimiter = $this->output_delimiter;
+        }
+
+        $data = $this->unparse($data, $fields, null, null, $delimiter);
+
+        if (!is_null($filename)) {
+            header('Content-type: application/csv');
+            header('Content-Length: ' . strlen($data));
+            header('Cache-Control: no-cache, must-revalidate');
+            header('Pragma: no-cache');
+            header('Expires: 0');
+            header('Content-Disposition: attachment; filename="' . $filename . '"; modification-date="' . date('r') . '";');
+
+            echo $data;
+        }
+
+        return $data;
+    }
+
+    /**
+     * Encoding
+     * Convert character encoding
+     *
+     * @access public
+     * @param  [string] $input  Input character encoding, uses default if left blank
+     * @param  [string] $output Output character encoding, uses default if left blank
+     */
+    public function encoding($input = null, $output = null) {
+        $this->convert_encoding = true;
+        if (!is_null($input)) {
+            $this->input_encoding = $input;
+        }
+
+        if (!is_null($output)) {
+            $this->output_encoding = $output;
+        }
+    }
+
+    /**
+     * Auto
+     * Auto-Detect Delimiter: Find delimiter by analyzing a specific number of
+     * rows to determine most probable delimiter character
+     *
+     * @access public
+     * @param  [string] $file         Local CSV file
+     * @param  [bool]   $parse        True/false parse file directly
+     * @param  [int]    $search_depth Number of rows to analyze
+     * @param  [string] $preferred    Preferred delimiter characters
+     * @param  [string] $enclosure    Enclosure character, default is double quote (").
+     *
+     * @return [string]
+     */
+    public function auto($file = null, $parse = true, $search_depth = null, $preferred = null, $enclosure = null) {
+        if (is_null($file)) {
+            $file = $this->file;
+        }
+
+        if (empty($search_depth)) {
+            $search_depth = $this->auto_depth;
+        }
+
+        if (is_null($enclosure)) {
+            $enclosure = $this->enclosure;
+        }
```

Comment:
This should have a `else { $this->enclosure = $enclosure; }` to correctly detect quote enclosures

Example:
`'"'`
`"'"`